### PR TITLE
Develop train

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -12,7 +12,7 @@ kotlin.version.snapshot=1.3-SNAPSHOT
 # Also set KONAN_LOCAL_DIST enviroment variable in bootstrap mode to auto-assign konan.home
 
 junit_version=4.12
-infra_version=0.1.0-dev-42
+infra_version=0.1.0-dev-45
```
Comment (sandwwraith) [inline gradle.properties]: There's a new `-dev-47` version which supports 1.3.30, 1.3.40 and master compilers

File: build.gradle [hunk 2]
```diff
@@ -91,4 +119,23 @@ subprojects {
 
 afterEvaluate {
     println("Using Kotlin compiler version: $org.jetbrains.kotlin.config.KotlinCompilerVersion.VERSION")
+    if (build_snapshot_train) {
```
Comment (sandwwraith) [inline build.gradle]: Maybe extract this to the separate file under `gradle` folder?

File: gradle/compilerVersion.gradle [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2017-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+gradle.taskGraph.whenReady {
```
Comment (qwwdfsad) [inline gradle/compilerVersion.gradle]: Please don't ask me to document this

File: build.gradle [hunk 4]
```diff
@@ -47,6 +66,7 @@ buildscript {
 }
 
 apply plugin: 'kotlinx.team.infra'
+ext.compilerVersion = org.jetbrains.kotlin.config.KotlinCompilerVersion.VERSION
```
Comment (sandwwraith) [inline build.gradle]: This is for infra or for some other reason? How this version differs from one printed in `compilerVersion.gradle`?

File: gradle.properties [hunk 5]
```diff
@@ -12,15 +12,15 @@ kotlin.version.snapshot=1.3-SNAPSHOT
...
 junit_version=4.12
-infra_version=0.1.0-dev-42
+infra_version=0.1.0-dev-47
 bintray_version=1.8.4-jetbrains-5
 benchmarks_version=0.1.7-dev-14
 
 native.deploy=false
 
-kotlin.parallel.tasks.in.project=true
+#kotlin.parallel.tasks.in.project=true
```
Comment (sandwwraith) [inline gradle.properties]: I think this should be uncommented for local development and disabled via system properties on TeamCity

File: build.gradle [hunk 6]
```diff
@@ -65,6 +85,15 @@ allprojects {
         version = version + '-SNAPSHOT'
     }
 
+    if (build_snapshot_train) {
+        // Snapshot-specific
+        def deployVersion = properties['DeployVersion']
```
Comment (sandwwraith) [inline build.gradle]: nit: `build_snapshot_train` and `kotlin_snapshot_version` are in snake case, but `DeployVersion` is in camel case. ¯\\\_(ツ)\_/¯

Comment (qwwdfsad) [reply build.gradle]: Yes, all our publication process are built around `DeployVersion`, I am not ready to rename it for the sake of this change.

File: build.gradle (hunk 4 above)
Comment (qwwdfsad) [reply build.gradle]: To make it visible from `compilerVersion.gradle` (FQN works only in root gradle file and imports are not transitive, yay)

**Pull request merged**